redislock.go

```diff
@@ -98,8 +124,16 @@ func (c *Client) Obtain(ctx context.Context, key string, ttl time.Duration, opt
 	}
 }
 
-func (c *Client) obtain(ctx context.Context, key, value string, ttl time.Duration) (bool, error) {
-	return c.client.SetNX(ctx, key, value, ttl).Result()
+func (c *Client) obtain(ctx context.Context, keys []string, value string, ttl time.Duration) (bool, error) {
+	setArgs := []string{}
+	for _, k := range keys {
+		setArgs = append(setArgs, k, value)
```

Comment:
Sorry, you cannot do that, you cannot pass `value` as redis keys. You need to do this:

```go
res, err := luaMSetNXEX.Run(ctx, c.client, keys, value, ttl.Milliseconds()).Result()
```

Reply:
Updated accordingly.